settings/settings-cpt.php

```diff
@@ -107,9 +107,16 @@ protected function form() {
 						printf(
 							'<li><label><input name="post_types[%s]" type="checkbox" value="1" %s %s/> %s</label></li>',
 							esc_attr( $post_type ),
-							checked( in_array( $post_type, $this->options['post_types'] ) || $disabled, true, false ),
+							checked( $disabled || in_array( $post_type, $this->options['post_types'] ), true, false ),
```

Comment:
The strict comparison was still missing.
```suggestion
							checked( $disabled || in_array( $post_type, $this->options['post_types'], true ), true, false ),
```